[CI] fix cardano-ledger-conformance build error

The CI has recently started failing this step; likely something to do with the new cost models stuff.

```
[ 6 of 33] Compiling Test.Cardano.Ledger.Conformance.SpecTranslate.Conway.Base ( src/Test/Cardano/Ledger/Conformance/SpecTranslate/Conway/Base.hs, /home/runner/work/formal-ledger-specifications/formal-ledger-specifications/cl/dist-newstyle/build/x86_64-linux/ghc-9.6.7/cardano-ledger-conformance-9.9.9.9/build/Test/Cardano/Ledger/Conformance/SpecTranslate/Conway/Base.o, /home/runner/work/formal-ledger-specifications/formal-ledger-specifications/cl/dist-newstyle/build/x86_64-linux/ghc-9.6.7/cardano-ledger-conformance-9.9.9.9/build/Test/Cardano/Ledger/Conformance/SpecTranslate/Conway/Base.dyn_o )

src/Test/Cardano/Ledger/Conformance/SpecTranslate/Conway/Base.hs:464:20: error: [GHC-83865]
    • Couldn't match type ‘()’ with ‘[((), ())]’
      Expected: SpecTransM ctx (Maybe [((), ())])
        Actual: SpecTransM
                  ctx
                  (SpecRep
                     (THKD
                        (Cardano.Ledger.Conway.PParams.PPGroups
                           Cardano.Ledger.Conway.PParams.TechnicalGroup
                           Cardano.Ledger.Conway.PParams.NoStakePoolGroup)
                        StrictMaybe
                        Cardano.Ledger.Plutus.CostModels.CostModels))
    • In a stmt of a 'do' block: ppuCostmdls <- toSpecRep cppCostModels
      In the expression:
        do ppuA <- toSpecRep cppTxFeePerByte
           ppuB <- toSpecRep cppTxFeeFixed
           ppuA0 <- toSpecRep cppA0
           ppuMinFeeRefScriptCoinsPerByte <- toSpecRep
                                               cppMinFeeRefScriptCostPerByte
           ....
      In an equation for ‘toSpecRep’:
          toSpecRep (ConwayPParams {..})
            = do ppuA <- toSpecRep cppTxFeePerByte
                 ppuB <- toSpecRep cppTxFeeFixed
                 ppuA0 <- toSpecRep cppA0
                 ....
    |
464 |     ppuCostmdls <- toSpecRep cppCostModels
    |                    ^^^^^^^^^^^^^^^^^^^^^^^
[ 9 of 33] Compiling Test.Cardano.Ledger.Conformance ( src/Test/Cardano/Ledger/Conformance.hs, /home/runner/work/formal-ledger-specifications/formal-ledger-specifications/cl/dist-newstyle/build/x86_64-linux/ghc-9.6.7/cardano-ledger-conformance-9.9.9.9/build/Test/Cardano/Ledger/Conformance.o, /home/runner/work/formal-ledger-specifications/formal-ledger-specifications/cl/dist-newstyle/build/x86_64-linux/ghc-9.6.7/cardano-ledger-conformance-9.9.9.9/build/Test/Cardano/Ledger/Conformance.dyn_o )
Error: [Cabal-7125]
Failed to build cardano-ledger-conformance-9.9.9.9 (which is required by test:tests from cardano-ledger-conformance-9.9.9.9).
```

--- a/cabal.project
+++ b/cabal.project
@@ -16,7 +16,7 @@ source-repository-package
   subdir: hs
   -- !WARNING!:
   -- MAKE SURE THIS POINTS TO A COMMIT IN `*-artifacts` BEFORE MERGE!
-  tag: 3da1ceec493ade9bc6347e342ca51054cc166e78
+  tag: ce149016c8294b9b91a9ee91fdbc449d6bb91a97
 
 source-repository-package
   type: git

--- a/libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/SpecTranslate/Conway/Base.hs
+++ b/libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/SpecTranslate/Conway/Base.hs
@@ -230,7 +230,7 @@ instance
       ppPv = (0, 0)
       ppMinUTxOValue = 0 -- minUTxOValue has been deprecated and is not supported in Conway
     ppCoinsPerUTxOByte <- toSpecRep cppCoinsPerUTxOByte
-    ppCostmdls <- toSpecRep cppCostModels
+    ppCostmdlsAssoc <- toSpecRep cppCostModels
     ppPrices <- toSpecRep cppPrices
     let
       pp = PParams cpp

--- a/libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/SpecTranslate/Core.hs
+++ b/libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/SpecTranslate/Core.hs
@@ -156,9 +156,9 @@ instance SpecTranslate ctx ProtVer where
   toSpecRep (ProtVer ver minor) = pure (getVersion ver, toInteger minor)
 
 instance SpecTranslate ctx CostModels where
-  type SpecRep CostModels = ()
+  type SpecRep CostModels = [((), ())]
 
-  toSpecRep _ = pure ()
+  toSpecRep _ = pure [((), ())]
 
 instance SpecTranslate ctx Prices where
   type SpecRep Prices = ()

